File: legacy-node/src/donorNeighbourhoodWindow.js
import { readEnvWithLegacy } from "./envLegacy.js";

const DEFAULT_HOURS = 2;
const MIN_HOURS = 1;
const MAX_HOURS = 72;

/**
 * @param {string | number | undefined} raw
 * @returns {number} whole hours between MIN_HOURS and MAX_HOURS
 */
export function parseDonorNeighbourhoodWindowHours(raw) {
  const parsed = Number(String(raw ?? DEFAULT_HOURS).trim());
  if (!Number.isFinite(parsed)) {
    return DEFAULT_HOURS;
  }
  return Math.min(MAX_HOURS, Math.max(MIN_HOURS, Math.round(parsed)));
}

/** From `INITIATOR_NEIGHBOURHOOD_WINDOW_HOURS` or legacy `DONOR_NEIGHBOURHOOD_WINDOW_HOURS` (default 2). */
export function getDonorNeighbourhoodWindowHours() {
  return parseDonorNeighbourhoodWindowHours(
    readEnvWithLegacy(
      "INITIATOR_NEIGHBOURHOOD_WINDOW_HOURS",
      "DONOR_NEIGHBOURHOOD_WINDOW_HOURS"
    )
  );
}

export function getDonorNeighbourhoodWindowMs() {
  return getDonorNeighbourhoodWindowHours() * 60 * 60 * 1000;
}

export function getDonorNeighbourhoodSinceQuery() {
  return `${getDonorNeighbourhoodWindowHours()}h`;
}

File: src/main/java/org/sharingbridge/integration/geo/DonorNeighbourhoodWindow.java
package org.sharingbridge.integration.geo;

import org.sharingbridge.integration.config.EnvLegacy;

public final class DonorNeighbourhoodWindow {

    public static final int DEFAULT_HOURS = 2;
    public static final int MIN_HOURS = 1;
    public static final int MAX_HOURS = 72;

    private DonorNeighbourhoodWindow() {}

    public static int parseDonorNeighbourhoodWindowHours(Object raw) {
        String text = raw == null ? String.valueOf(DEFAULT_HOURS) : String.valueOf(raw).trim();
        try {
            double parsed = Double.parseDouble(text);
            if (!Double.isFinite(parsed)) {
                return DEFAULT_HOURS;
            }
            return Math.min(MAX_HOURS, Math.max(MIN_HOURS, (int) Math.round(parsed)));
        } catch (NumberFormatException ex) {
            return DEFAULT_HOURS;
        }
    }

    public static int getDonorNeighbourhoodWindowHours() {
        return parseDonorNeighbourhoodWindowHours(
                EnvLegacy.readEnvWithLegacy(
                        "INITIATOR_NEIGHBOURHOOD_WINDOW_HOURS",
                        "DONOR_NEIGHBOURHOOD_WINDOW_HOURS"));
    }

    public static long getDonorNeighbourhoodWindowMs() {
        return getDonorNeighbourhoodWindowHours() * 60L * 60L * 1000L;
    }

    public static String getDonorNeighbourhoodSinceQuery() {
        return getDonorNeighbourhoodWindowHours() + "h";
    }
}
